procrastinate/sql/schema.sql

```diff
@@ -269,12 +269,30 @@ DECLARE
     _job_id bigint;
 BEGIN
     UPDATE procrastinate_jobs
-    SET status = 'todo',
-        attempts = attempts + 1,
-        scheduled_at = retry_at,
-        priority = COALESCE(new_priority, priority),
-        queue_name = COALESCE(new_queue_name, queue_name),
-        lock = COALESCE(new_lock, lock)
+    SET status = CASE
+            WHEN abort_requested = false THEN 'todo'::procrastinate_job_status
+            ELSE 'failed'::procrastinate_job_status
```

Comment:
How about  just using NOT:
```sql
WHEN NOT abort_requested THEN 'todo'::procrastinate_job_status
ELSE 'failed'::procrastinate_job_status
```
or maybe even just the other way around
```sql
WHEN abort_requested THEN 'failed'::procrastinate_job_status
ELSE 'todo'::procrastinate_job_status
```
Same for other cases and migration.

Reply:
We can invert the condition.
The reason I went with this order is that the first when represents the normal case on retry.